src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -142,6 +147,21 @@ bool IRequest.WaitForCompletion(int timeout)
         }
 
         /// <summary>
+        /// Gets TestCaseFilterExpression from discovery criteria's test case filter
+        /// </summary>
+        private TestCaseFilterExpression GetTestCaseFilter()
+        {
+            FilterExpressionWrapper filterExpressionWrapper = !string.IsNullOrEmpty(this.DiscoveryCriteria.TestCaseFilter) ? new FilterExpressionWrapper(this.DiscoveryCriteria.TestCaseFilter) : null;
+            return (filterExpressionWrapper != null && string.IsNullOrEmpty(filterExpressionWrapper.ParseError)) ?
+                new TestCaseFilterExpression(filterExpressionWrapper) : null;
+        }
```

Comment:
Can't we do this in the logger end itself. As the entire discovery criteria is already passed as part of the DiscoveryStartEventArgs ?

Reply:
FilterExpressionWrapper and TestCaseFilterExpression is not exposed to logger. Only ITestCaseFilterExpression interface is exposed and thus cant be done at logger side